### PR TITLE
Improve eval docs

### PR DESCRIPTION
**Pull request by roberth**

### CODE REVIEW

**Comment by domenkozar** [comment]
This should be narrative documentation under `Evaluation on agents` or something, seems weird to put everything in an FAQ

**Comment by roberth** [comment]
I agree but it's the best place we have for now.
Could you open an issue describing the type of documentation you would like to see this moved into? I'd do it but I'm not sure it's what you want and I have to leave :)

File: docs/modules/reference/pages/evaluation.adoc [hunk 1]
```diff
@@ -9,3 +9,13 @@ Evaluation is the first step when it comes to building your project. It is perfo
     search path. This is to ensure reproducability and purity also when it comes to nixpkgs
 . If any of the attributes fail to evaluate, job evaluation phase will be marked as failed
    and error messages will be shown inline per attribute
+
```
Comment (domenkozar) [inline docs/modules/reference/pages/evaluation.adoc]:
```suggestion
== Differences from nix-build
```

File: docs/modules/reference/pages/evaluation.adoc [hunk 2]
```diff
@@ -9,3 +9,13 @@ Evaluation is the first step when it comes to building your project. It is perfo
...
+The agent expects the root of the file to be an attribute set and traverses it similarly to how `nix build` does it.
+A few differences exists:
+
+* All attribute names are allowed, including ones with characters like `.` that are ignored by `nix build`
+* The root must not be a list
+
+NOTE: like `nix-build`, the evaluator will only traverse a nested attribute set if
+the nested attribute set itself has an attribute `recurseForDerivations = true`.
+This can be set with the `pkgs.recurseIntoAttrs` function.
```
Comment (domenkozar) [inline docs/modules/reference/pages/evaluation.adoc]: A small example would be better than telling which function to use.

File: docs/modules/reference/pages/evaluation.adoc [hunk 3]
```diff
@@ -9,3 +9,13 @@ Evaluation is the first step when it comes to building your project. It is perfo
     search path. This is to ensure reproducability and purity also when it comes to nixpkgs
 . If any of the attributes fail to evaluate, job evaluation phase will be marked as failed
    and error messages will be shown inline per attribute
+== Differences from nix-build
```
Comment (roberth) [inline docs/modules/reference/pages/evaluation.adoc]:
```suggestion

== Differences from nix-build

```

**Pull request merged**